File: app/src/main/java/com/eam/investire/activity/MainActivity.java
package com.eam.investire.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.eam.investire.R;
import com.eam.investire.config.ConfiguracaoFirebase;
import com.google.firebase.auth.FirebaseAuth;
import com.heinrichreimersoftware.materialintro.app.IntroActivity;
import com.heinrichreimersoftware.materialintro.slide.FragmentSlide;

public class MainActivity extends IntroActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setButtonBackVisible(false);
        setButtonNextVisible(false);

        addSlide(new FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_1)
                .build());

        addSlide(new FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_2)
                .build());

        addSlide(new FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_3)
                .build());

        addSlide(new FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_4)
                .build());

        addSlide(new FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_cadastro)
                .canGoForward(false)
                .build());
    }

    @Override
    protected void onStart() {
        super.onStart();
        verificarUsuarioLogado();
    }

    public void btEntrar(View view) {
        startActivity(new Intent(this, LoginActivity.class));
    }

    public void btCadastrar(View view) {
        startActivity(new Intent(this, CadastroActivity.class));
    }

    public void verificarUsuarioLogado() {
        FirebaseAuth autenticacao = ConfiguracaoFirebase.getFirebaseAutenticacao();
        if (autenticacao.getCurrentUser() != null) {
            abrirTelaPrincipal();
        }
    }

    public void abrirTelaPrincipal() {
        startActivity(new Intent(this, PrincipalActivity.class));
    }
}

File: app/src/main/java/com/eam/investire/activity/MainActivity.kt
package com.eam.investire.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.eam.investire.R
import com.eam.investire.config.ConfiguracaoFirebase
import com.heinrichreimersoftware.materialintro.app.IntroActivity
import com.heinrichreimersoftware.materialintro.slide.FragmentSlide

class MainActivity : IntroActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        isButtonBackVisible = false
        isButtonNextVisible = false
        addSlide(FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_1)
                .build())
        addSlide(FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_2)
                .build())
        addSlide(FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_3)
                .build())
        addSlide(FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_4)
                .build())
        addSlide(FragmentSlide.Builder()
                .background(R.color.colorGray)
                .fragment(R.layout.intro_cadastro)
                .canGoForward(false)
                .build())
    }

    override fun onStart() {
        super.onStart()
        verificarUsuarioLogado()
    }

    fun btEntrar(view: View?) {
        startActivity(Intent(this, LoginActivity::class.java))
    }

    fun btCadastrar(view: View?) {
        startActivity(Intent(this, CadastroActivity::class.java))
    }

    private fun verificarUsuarioLogado() {
        val autenticacao = ConfiguracaoFirebase.firebaseAutenticacao
        if (autenticacao!!.currentUser != null) {
            abrirTelaPrincipal()
        }
    }

    private fun abrirTelaPrincipal() {
        startActivity(Intent(this, PrincipalActivity::class.java))
    }
}